source/solvers/heat_transfer.cc

```diff
@@ -40,6 +43,142 @@ void
 HeatTransfer<dim>::assemble_rhs()
 {
   assemble_system_rhs();
+  if (this->simulation_parameters.nitsche->number_solids > 0)
+    assemble_constraints(true);
+}
+
+template <int dim>
+void
+HeatTransfer<dim>::assemble_constraints(bool assemble_matrix)
+{
+  Assert(
+    !this->simulation_parameters.physical_properties_manager.is_non_newtonian(),
+    RequiresConstantViscosity("assemble_nitsche_restriction"));
+
+  // Evaluate fluid properties
+  auto density_model =
+    this->simulation_parameters.physical_properties_manager.get_density();
+  auto specific_heat_model =
+    this->simulation_parameters.physical_properties_manager.get_specific_heat();
+  auto conductivity_model =
+    this->simulation_parameters.physical_properties_manager
+      .get_thermal_conductivity();
+  std::map<field, double> field_values;
+
+  double rho_cp = density_model->value(field_values) *
+                  specific_heat_model->value(field_values);
+
+  auto solids = *this->multiphysics->get_solids();
+
+  // Loops over solids
+  for (unsigned int i_solid = 0; i_solid < solids.size(); ++i_solid)
+    {
+      // Initialize the solid and its particles
+      std::shared_ptr<Particles::ParticleHandler<dim>> &solid_ph =
+        solids[i_solid]->get_solid_particle_handler();
+
+      if (this->simulation_parameters.nitsche->nitsche_solids[i_solid]
+            ->enable_heat_bc)
+        {
+          const unsigned int dofs_per_cell = this->fe->dofs_per_cell;
+
+          std::vector<types::global_dof_index> heat_dof_indices(dofs_per_cell);
+          FullMatrix<double>     local_matrix(dofs_per_cell, dofs_per_cell);
+          dealii::Vector<double> local_rhs(dofs_per_cell);
+
+          Function<dim> *solid_temperature =
+            solids[i_solid]->get_solid_temperature();
+
+          // Penalization terms
+          const double beta =
+            this->simulation_parameters.nitsche->nitsche_solids[i_solid]
+              ->beta_heat;
+
+          // Loop over all local particles
+          auto particle = solid_ph->begin();
+          while (particle != solid_ph->end())
+            {
+              local_matrix = 0;
+              local_rhs    = 0;
+
+#if (DEAL_II_VERSION_MAJOR < 10 && DEAL_II_VERSION_MINOR < 4)
+              const auto &cell =
+                particle->get_surrounding_cell(*this->triangulation);
+#else
+              const auto &cell = particle->get_surrounding_cell();
+#endif
```

Comment:
We don,t support 9.3 anymore, this shouldnt be needed.
```suggestion
              const auto &cell = particle->get_surrounding_cell();

```